component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java

```diff
@@ -1825,10 +1834,16 @@ public static SiteEntity buildSiteEntity(PortalConfig site,
                                              getLayoutService(),
                                              getUserACL());
     }
-    return new SiteEntity(siteType,
+    PortalConfig sitePortalConfig = getLayoutService().getPortalConfig(new SiteKey(siteType, site.getName()));
+    long siteId = Long.parseLong((sitePortalConfig.getStorageId().split("_"))[1]);
+    String translatedTitle = getTranslatedLabel(SITE_TITLE_FIELD_NAME, siteId, locale);
+    String translatedDescription= getTranslatedLabel(SITE_DESCRIPTION_FIELD_NAME, siteId, locale);
+    displayName = StringUtils.isNotBlank(translatedTitle) ? translatedTitle : displayName;
```

Comment:
```suggestion
    String displayName = StringUtils.isNotBlank(translatedTitle) ? translatedTitle : site.getLabel();
```
Remove the first instruction String displayName = site.getLabel(); above